meilisearch-http/src/index/search.rs

```diff
@@ -203,8 +201,138 @@ impl Index {
     }
 }
 
+fn compute_formatted_options(
+    attr_to_highlight: &HashSet<String>,
+    attr_to_crop: &[String],
+    query_crop_length: usize,
+    to_retrieve_ids: &BTreeSet<u8>,
+    fields_ids_map: &FieldsIdsMap,
+    displayed_ids: &BTreeSet<u8>,
+    ) -> BTreeMap<FieldId, FormatOptions> {
+
+    let mut formatted_options = BTreeMap::new();
+
+    formatted_options = add_highlight_to_formatted_options(
+        formatted_options,
+        attr_to_highlight,
+        fields_ids_map,
+        displayed_ids,
+    );
+
+    formatted_options = add_crop_to_formatted_options(
+        formatted_options,
+        attr_to_crop,
+        query_crop_length,
+        fields_ids_map,
+        displayed_ids,
+    );
+
+    // Should not return `_formatted` if no valid attributes to highlight/crop
+    if !formatted_options.is_empty() {
+        formatted_options = add_non_formatted_ids_to_formatted_options(
+            formatted_options,
+            to_retrieve_ids,
+        );
+    }
+
+    formatted_options
+}
+
+fn add_highlight_to_formatted_options(
+    mut formatted_options: BTreeMap<FieldId, FormatOptions>,
+    attr_to_highlight: &HashSet<String>,
+    fields_ids_map: &FieldsIdsMap,
+    displayed_ids: &BTreeSet<u8>,
+) -> BTreeMap<FieldId, FormatOptions> {
+    for attr in attr_to_highlight {
+        let new_format = FormatOptions {
+            highlight: true,
+            crop: None,
+        };
+
+        if attr == "*" {
+            for id in displayed_ids {
+                formatted_options.insert(*id, new_format);
+            }
+            break;
+        }
+
+        if let Some(id) = fields_ids_map.id(&attr) {
+            if displayed_ids.contains(&id) {
+                formatted_options.insert(id, new_format);
+            }
+        }
+    }
+
+    formatted_options
+}
+
+fn add_crop_to_formatted_options(
+    mut formatted_options: BTreeMap<FieldId, FormatOptions>,
+    attr_to_crop: &[String],
+    crop_length: usize,
+    fields_ids_map: &FieldsIdsMap,
+    displayed_ids: &BTreeSet<u8>,
+) -> BTreeMap<FieldId, FormatOptions> {
+    for attr in attr_to_crop {
+        let mut attr_name = attr.clone();
+        let mut attr_len = crop_length;
+
+        let mut split = attr_name.rsplitn(2, ':');
+        attr_name = match split.next().zip(split.next()) {
+            Some((len, name)) => {
+                attr_len = len.parse().unwrap_or(crop_length);
+                name.to_string()
+            },
+            None => attr_name,
+        };
+
+        if attr_name == "*" {
+            for id in displayed_ids {
+                formatted_options
+                    .entry(*id)
+                    .and_modify(|f| f.crop = Some(attr_len))
+                    .or_insert(FormatOptions {
+                        highlight: false,
+                        crop: Some(attr_len),
+                    });
+            }
```

Comment:
```suggestion
            }
            break
```

Reply:
I don't want to break because we have to let the users does: `"attributesToCrop": ["*", "author:10"]`
To say "all the attributes have to be cropped with the default size, but author has to be cropped with a size of 10"
https://docs.meilisearch.com/reference/features/search_parameters.html#attributes-to-crop